controllers/controllers/networking/cfroute_controller.go

```diff
@@ -188,17 +192,19 @@ func (r *CFRouteReconciler) finalizeCFRoute(ctx context.Context, cfRoute *korifi
 		return ctrl.Result{}, nil
 	}
 
-	fqdnHTTPProxy, foundFQDNProxy, err := r.getFQDNProxy(ctx, cfRoute.Spec.Host, cfDomain.Spec.Name, cfRoute.Namespace, false)
-	if err != nil {
-		return ctrl.Result{}, err
-	}
-
-	// Cleanup the FQDN HTTPProxy on delete
-	if foundFQDNProxy {
-		err := r.finalizeFQDNProxy(ctx, cfRoute.Name, fqdnHTTPProxy)
+	if cfDomain != nil {
```

Comment:
A few questions:
- Do we need `cfDomain` here? It looks like `finalizeFQDNProxy` could just get `cfRoute.Status.FQDN`.
- If no, can we get rid of this `if` and always finalise the FQDN proxy?
- If yes, can we call `finalizeCFRoute` even before trying to retrieve the domain, right after retrieving the route?

Reply:
@gcapizzi thanks, that's a good suggestion. I was thinking we needed the domain in order to determine the fqdn, but you are right that we already have that information denormalized into the route status.

I will give that change a try.

Reply:
I assume if it is missing in the route status then we haven't reconciled the `CFRoute` and there is no way we're potentially missing an `HTTPProxy` we need to update?